.github/workflows/pytest.yml

```diff
@@ -172,19 +173,36 @@ jobs:
     defaults:
       run:
         shell: bash -l {0}
-
     steps:
+      - name: Collect Workflow Telemetry
+        uses: catchpoint/workflow-telemetry-action@v2
```

Comment:
This generates plots of CPU usage, RAM usage, Disk I/O, Network I/O. Unfortunately, if the runner gets summarily shut down they aren't generated. Also unfortunately, it doesn't show disk usage.